hack/tools/go.mod

```diff
@@ -7,6 +7,7 @@ go 1.22.5
 
 require (
 	github.com/d2iq-labs/helm-list-images v0.11.0
+	golang.org/x/mod v0.17.0
```

Comment:
nit: The rest of the code uses `"github.com/blang/semver/v4` would that package work here?

Reply:
+1, if possible

Reply:
I don't see this being pulled directly in the go.mod for tools: https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/939/files#diff-f389ca9ba468c8313b9d514e887e4b8899886bb2d6790352386aaf01da3aa588R8-R16

for the api, I chose golang.org/x/mod/semver to avoid adding more third-party dependencies for consumers as golang.org/x/mod/semver is technically still upstream go.

Reply:
okay; done. Although, I liked the APIs on the older package better.